build.rs

```diff
@@ -574,10 +689,27 @@ fn main() -> Result<()> {
     // If you have troubles with your shell accepting '.' in a var name,
     // try using 'env' (sorry but this isn't our fault - it just has to
     // match the pkg-config package name, which is going to have a . in it).
-    let cross_compiling =
-        env::var("PYO3_CROSS_INCLUDE_DIR").is_ok() && env::var("PYO3_CROSS_LIB_DIR").is_ok();
+    //
+    // Detecting if cross-compiling by checking if the target triple is different from the host
+    // rustc's triple.
+    let cross_compiling = env::var("TARGET") != env::var("HOST");
```

Comment:
It's nice to have this check, but the condition might need to be different. 

E.g. this looks like it'll break 32-bit builds on windows (where the host rustc is usually 64-bit).

Reply:
I did not know that detail! i'll look into what needs to be done here in such a case. Maybe looking at how other popular crates detect cross compilation.

Reply:
Are you refering to when building a 32bit windows binary with a 64bit windows machine? if so the target should be different as according to [rust platform support] windows 32 bit target start with `i686` where as a 64bit windows should have `x86_64`.

Unfortunately I dont have access to windows with a development environment to test this out... However, by inspecting the [openssl-sys] crates [build] seems to basically boil down to this same comparison

[openssl-sys]: https://docs.rs/openssl-sys/0.9.58/openssl_sys/
[build]: https://github.com/sfackler/rust-openssl/blob/master/openssl-sys/build/find_normal.rs#L24
[rust platform support]: https://forge.rust-lang.org/release/platform-support.html

Reply:
Yep, what I mean though is that a 32-bit windows Python interpreter is still available for us to execute in the normal fashion, so we can get the config from the Python interpreter. Especially as we now know we can't use the `sysconfigdata` file on Windows, we definitely need to go down the "not cross compiling" configuration route.

I think if you rebase on master (where I've made a couple of CI fixes), the 32-bit windows builds will fail, which'll show what I mean 👍

Reply:
i'll check it out. it'll help since i dont have access to a windows dev machine

Reply:
[seems to pass](https://github.com/PyO3/pyo3/pull/1095/checks?check_run_id=994290675)

Reply:
Nice, happy to see I'm wrong!